### PR TITLE
feat: add `spacy.WandbLogger.v3` with optional `run_name` and `entity` parameters

### PR DESCRIPTION
**Pull request by Joozty**
## Description

Just adding `run_name` to WandbLogger. When you run multiple trainings sometimes it's better to assign them own name so it's easier to distinguish them in Wandb Dashboard

https://docs.wandb.ai/guides/track/launch#can-i-just-set-the-run-name-to-the-run-id

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [ ] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Review by svlandeg — CHANGES_REQUESTED** [review]
Thanks for the contribution! We will definitely consider merging this, but we'd want to bump the logger's version number to v3 (a copy of v2 can stay in the code for now, we'll clean it up later). And perhaps then we could also include the `entity_name` addition from https://github.com/explosion/spaCy/pull/8832?

**Comment by Joozty** [comment]
Is bumping logger necessary? `run_name` is an optional parameter and it doesn't break anything. `entity_name` can be an optional parameter as well according to [docs](https://docs.wandb.ai/ref/python/init).

**Comment by svlandeg** [comment]
It wouldn't break backwards compatibility if we didn't bump, but it would still potentially create problems: a config with a v2 in it, created with a new spaCy version and specifying the new parameter(s), suddenly wouldn't run anymore on an older version of spaCy that didn't have the new parameters yet, it would raise an error about the parameter being unavailable, which isn't what you expect if you've used the exact same version of the logger.

**Comment by Joozty** [comment]
Oh I see, thank 🙂 Should I create v3 then?

**Comment by svlandeg** [comment]
Yes, please :-) for now you can just copy-paste the code and keep one copy as v2, and make your changes in the other copy as v3.

File: website/docs/api/top-level.md [hunk 1]
```diff
@@ -507,6 +507,8 @@ remain in the config file stored on your local system.
 | `remove_config_values` | A list of values to include from the config before it is uploaded to W&B (default: empty). ~~List[str]~~                              |
 | `model_log_interval`   | Steps to wait between logging model checkpoints to W&B dasboard (default: None). ~~Optional[int]~~                                    |
 | `log_dataset_dir`      | Directory containing dataset to be logged and versioned as W&B artifact (default: None). ~~Optional[str]~~                            |
+| `run_name`             | The name of the run. If you don't specify a run_name, the name will be created by wandb library. (default: None ). ~~Optional[str]~~                            |
+| `entity`               | An entity is a username or team name where you're sending runs. If you don't specify an entity, the run will be sent to your default entity, which is usually your username.  (default: None). ~~Optional[str]~~                            |
```
Comment (Joozty) [inline website/docs/api/top-level.md:511]: I copied these bits for [official documentation](https://docs.wandb.ai/ref/python/init). Maybe I should somehow reference it? Not sure what's the best way to do it. 🤔

Comment (svlandeg) [reply website/docs/api/top-level.md:511]: I think this is fine!

**Review by svlandeg — APPROVED** [review]

**Comment by adrianeboyd** [comment]
I think we should accept this PR in the upcoming [`spacy-loggers`](https://github.com/explosion/spacy-loggers) repo rather than merging it here at all. It looks like this has already been copied over, but I haven't double-checked the code/tests.

**Comment by svlandeg** [comment]
We've discussed a little bit more internally, and we'll still merge this for v.3.1.3.
From 3.2 onwards, we'll be using the separate repo `spacy-loggers` as Adriane noted, and from then on the code will be moved to there.

Thanks again for this contribution!

**Pull request merged**